util/importers/ldc93s1.py

```diff
@@ -0,0 +1,155 @@
+        for batch_thread in batch_threads:
+            batch_thread.daemon = True
+            batch_thread.start()
+
+    def _create_files_circular_list(self):
+        priorityQueue = PriorityQueue()
+        for txt_file in self._txt_files:
+          wav_file = path.splitext(txt_file)[0] + ".wav"
+          wav_file_size = getsize(wav_file)
+          priorityQueue.put((wav_file_size, (txt_file, wav_file)))
+        files_list = []
+        while not priorityQueue.empty():
+            priority, (txt_file, wav_file) = priorityQueue.get()
+            files_list.append((txt_file, wav_file))
+        return cycle(files_list)
+
+    def _populate_batch_queue(self):
+        with self._graph.as_default():
+            while True:
+                n_steps = 0
+                sources = []
+                targets = []
+                for index, (txt_file, wav_file) in enumerate(self._files_circular_list):
+                    if index >= self._batch_size:
+                        break
+                    next_source = audiofile_to_input_vector(wav_file, self._numcep, self._numcontext)
+                    if n_steps < next_source.shape[0]:
+                        n_steps = next_source.shape[0]
+                    sources.append(next_source)
+                    with open(txt_file) as open_txt_file:
+                        original = ' '.join(open_txt_file.read().strip().lower().split(' ')[2:]).replace('.', '')
+                        targets.append(original)
+                target = texts_to_sparse_tensor(targets)
+                for index, next_source in enumerate(sources):
+                    npad = ((0,(n_steps - next_source.shape[0])), (0,0))
+                    sources[index] = np.pad(next_source, pad_width=npad, mode='constant')
+                source = np.array(sources)
+                self._batch_queue.put((source, target))
+
+    def next_batch(self):
+        source, target = self._batch_queue.get()
+        return (source, target, source.shape[1])
+
+    @property
+    def total_batches(self):
+        # Note: If len(_txt_files) % _batch_size != 0, this re-uses initial _txt_files
+        return int(ceil(float(len(self._txt_files)) /float(self._batch_size)))
+
+
+def read_data_sets(graph, data_dir, batch_size, numcep, numcontext, thread_count=1):
+    # Conditionally download data
+    LDC93S1_BASE = "LDC93S1"
+    LDC93S1_BASE_URL = "https://catalog.ldc.upenn.edu/desc/addenda/"
+    local_file = base.maybe_download(LDC93S1_BASE + ".wav", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".wav")
+    _ = base.maybe_download(LDC93S1_BASE + ".txt", data_dir, LDC93S1_BASE_URL + LDC93S1_BASE + ".txt")
+
+    # Conditionally extract LDC93S1 data
+    LDC93S1_DIR = "LDC93S1"
+    _maybe_extract(data_dir, LDC93S1_DIR, local_file, ["dev", "test", "train"])
+
+    # Create dev DataSet
+    train = dev = test = _read_data_set(graph, data_dir, LDC93S1_DIR, "dev", thread_count, batch_size, numcep, numcontext)
+
+    # Create test DataSet
```

Comment:
Could you remove these commented out lines or un-comment them and have the line above only assign to `dev`